daemon/daemon.go

```diff
@@ -482,14 +529,47 @@ func (d *Daemon) Stop() error {
 		d.snapListener.Close()
 	}
 
+	if restartSystem {
+		// give time to polling clients to notice restart
+		time.Sleep(300 * time.Millisecond)
+	}
+
 	d.tomb.Kill(d.snapdServe.finishShutdown())
 	if d.snapListener != nil {
 		d.tomb.Kill(d.snapServe.finishShutdown())
 	}
 
+	if !restartSystem {
+		// tell systemd that we are stopping
+		systemdSdNotify("STOPPING=1")
+
+	}
+
 	d.overlord.Stop()
 
-	return d.tomb.Wait()
+	err := d.tomb.Wait()
+	if err != nil {
+		return err
+	}
+
+	if restartSystem {
+		// ask for shutdown and wait for it to happen.
```

Comment:
Should we care about any logged in users at this point? To give them time to read the message that the system is going down etc?

Reply:
I had a conversation about this at the last sprint with Tony, that's why now I do reboot with "+1", not "now"